src/plugins/data_views/common/fields/data_view_field.ts

```diff
@@ -108,13 +108,29 @@ export class DataViewField implements IFieldType {
   }
 
   public get type() {
-    return this.runtimeField?.type
-      ? castEsToKbnFieldTypeName(this.runtimeField?.type)
-      : this.spec.type;
+    if (this.isRuntimeField) {
+      let type: RuntimeType = this.runtimeField?.type!;
+      if (this.isRuntimeCompositeSubField) {
+        const [, subFieldName] = this.name.split('.');
```

Comment:
I can currently have a composite field name `composite_name.a` and inside `emit('a','a');emit('b','b')` 

I think this `.split` check won't work currently in this case

Reply:
Just for illustration the scenario where this check would work incorrectly: 


![Screen Shot 2022-02-11 at 17 21 37](https://user-images.githubusercontent.com/7784120/153628781-9e3484d6-ba59-4d53-8536-57a25bf602e9.png)



maybe the best fix would be for composite runtime field have it' original name available at `this.runtimeField.name`

Reply:
Hm, its pretty clear that a bunch of this code needs to be rethought

Reply:
Now type is set on a runtime field when the field list is created or updated.